core/block.go

```diff
@@ -402,7 +402,7 @@ func UnmarshalBlockNumber(val []byte) uint64 {
 	return binary.BigEndian.Uint64(val)
 }
 
-func concatCounts(txCount, eventCount, stateDiffLen uint64, l1Mode L1DAMode) *felt.Felt {
+func ConcatCounts(txCount, eventCount, stateDiffLen uint64, l1Mode L1DAMode) *felt.Felt {
```

Comment:
Oh this function returns a pointer, it should return by value. How big is the part of the code that gets affected by changing this to return by value. If it's just a couple of line I would hope we can make it return by value. Otherwise we open an issue with it